flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/Buckets.java

```diff
@@ -187,6 +187,10 @@ private void handleRestoredBucketState(final BucketState<BucketID> recoveredStat
 
 	private void updateActiveBucketId(final BucketID bucketId, final Bucket<IN, BucketID> restoredBucket) throws IOException {
 		if (!restoredBucket.isActive()) {
+			if (bucketLifeCycleListener != null) {
```

Comment:
Add a method: `inactiveBucket(Bucket<IN, BucketID>)`?

Reply:
Extracted the common `notifyBucketCreate` and `notifyBucketInactive` method.